 \begin{abstract}

We simulated RNA-Seq count data based on parameters estimated from a maize RNA-Seq dataset \citep{paschold2012complementation}. We comprehensively compared the newly proposed empirical Bayesian differential expression (DE) analysis method, denoted as {\tt eBayes},  to the alternative five DE analysis methods implemented by the existing R packages {\tt edgeR, DESeq2, DESeq, sSeq, EBSeq}. We evaluated their performance by receiver operator characteristic (ROC) curves and areas under the curve (AUC). {\tt eBayes} seems to have the best performance in terms of AUC. We observed the following patterns: (1) the difference among methods shrinks as proportion of DE genes (pDiff) increases; (2) the number of genes (nGenes) doesn't affect the methods performance in terms of AUC values; (3) all methods perform better when the number of samples increases. Supplementary materials accompanying this paper is on github at https://github.com/xiyuansun/kellycc. 

 \end{abstract}
