 \begin{abstract}

We simulated RNA-Seq count data based on parameters estimated from a maize RNA-Seq dataset \citep{paschold2012complementation}. We comperehensively compared six differential expression (DE) analysis methods (eBayes, edgeR, DESeq2, DESeq, sSeq, and EBSeq) and evaluate their performance by receiver operator characteristic (ROC) curves and areas under the curve (AUC). eBayes tends to give the best performance in terms of AUC. We observed the following patterns: (1) the difference among methods shrank as proportion of DE genes (pDiff) increases; (2) the number of genes (nGenes) doesn't affect the methods performance in terms of AUC values; (3) all methods perform better when the number of samples increases.Supplementary materials accompanying this paper is on github at https://github.com/xiyuansun/kellycc. 

 \end{abstract}
