Differential gene expression (DGE) refers to the difference in the gene expression between two or more varieties based on read counts from replicated samples. An approach for comparing gene expression levels between replicated conditions of RNA sequencing data replies on counting reads that map to features of insterest. Within such count-based methods, many flexible and advanced statistical approaches now exist and offer the ability to adjust for covariates (e.g., batch effects) \cite{zhou2014robustly}. often, these methods include some sort of sharing of information across features to improve inferences in small samples. Jarad developed a hierarchical negative binomial model and drew inferences using a computationally tractable empirical Bayes approach to inference \cite{niemi2015empirical}. In this report, I demonstrate improvements over alternative methods via a simulation study based on a maize experiment. This article has supplementary material online \href{https://github.com/jarad/eBayes_differential_expression}{GitHub repo for this report} 