% Introduction of the Thesis Template File
\chapter{OVERVIEW}



\section{Introduction}

Differential expression exists when the expected value of a phenotype differs from the expected phenotypic values of the other variety. Differential expression can occue if the mean phenotype of a variety is greater than the other variety's or less than the other variety's. I refer to the former as high differential expression (HDE) and the latter as low differential expression (LDE). There is differential expression (DE) if and only if either HDE or LDE holds. Ji et al. \cite{ji2014estimation} introduced an approach to assess gene expression heterosis using microarray data under the assumption that these data are continuous. They built a normal hierarchical model for microarray measurements of transcript abundance that allows borrowing of information across genes to estimate means and variances. They introduced an empirical Bayes framework that first estimates model hyperparameters, then estimates the posterior distribution for gene-specific parameters conditional on those hyperparameters, and finally computes heterosis probabilities based on integrals of regions under this posterior. Building on the work of Ji et al. with the normal data model, Niemi et al. \cite{niemi2015empirical} constructed a hierarchical model based on a negative binomial data model. They utilized an empirical Bayes approach to obtain estimates of the hyperparameters and the posterior distributions for the gene-specific parameters conditional on those hyperparameters. In this report, I applied Niemi's empirical Bayes method in the differential gene expression analysis context. 


The remainder of the report proceeds as follows.Chapter 2 presents the hierarchical model, an empirical Bayes method of estimating the parameters, and the calculation of posterior probabilities of DE. Chapter 3 presents a simulation study based on a maize experiment and compares Niemi et al's approach to alternative methods. Chapter 4 summarizes the work and suggests directions for future research.


\section{RNA-Seq Count Data}
The focus here is on methods for count-based differential expression (DE) amalyses. Thus, the starting point here is a count table of features-by-samples, such as those from the Paschold project \cite{paschold2012complementation}.

The number of reads mapped to a given gene is considered to be the estimate of the expression level of that feature using the technology. The end product of a RNA-seq experiment is a sequence of read counts, typically represented as a matrix with rows representing genes and columns representing samples from two varieties, as in Table \ref{tab:RNA-Seq Data}. In this example, there are $V=2$ varieties, $J_1 = 2$ samples in the first variety, $J_2=2$ samples in the second variety, and $G=10000$ genes. My interest is in the detection of differentially expressed genes among the varieties. 

\begin{table}[H]
\begin{center}
    \begin{tabular}{|c|c|c|c|c|c|}
      \hline
      Gene &Variety1 Sample1 &Variety1 Sample2 &Variety2 Sample1 & Variety2 Sample2 \\
      \hline
      1 & 4 & 1 & 100 & 88 \\
      \hline
      2 & 65 & 48 & 55 & 59 \\
      \hline
      3 & 0 & 1 & 0 & 2\\
      \hline
      ... & ... & ... & ... & ...\\
      \hline
       10000 & 3 & 1 & 1 & 2\\
       \hline
    \end{tabular}
\end{center}
\caption{RNA-Seq Count Data Example}
\label{tab:RNA-Seq Data}
\end{table}

The focus of this report is to provide a comparison of the methods related to the analysis of differential expression for RNA-seq data. In this report, I review statistical methods for detecting differential expression in the RNA-seq data, including the empirical Bayesian method. I summarize the results of a simulation study. I briefly describe some existing open source R and Bioconductor software for testing differential expression for RNA-seq data. I conclude the report with a discussion section. 

