\chapter{Simulation Study}

\section{Simulation Study based on a maize experiment}

We built a simulation framework that aims to reflect the reality of RNA-seq data. 

\subsection{Parameter Estimation}

To assess the efficacy of \texttt{eBayes} method to identify DE genes, we used a maize dataset with varieties $B73$ and $Mo17$ \citep{paschold2012complementation} to determine realistic parameter values for our simulation study. Chapter 2 describes the maize dataset in detail. 

We removed the genes with zero counts in all conditions, as well as genes whose maximum counts are less than $5$ as recommended \citep{rau2013data}.The description of parameters for the real RNA-seq dataset is summarized in Table \ref{tab:Parameter-Estimation}. 

\begin{table}[ht]
\centering
\begin{tabular}{|r|r|}
\hline
Number of Trimmed Genes & 27619 \\ 
\hline
Number of Samples & 8\\
\hline
  Median expression ($\log{2}$ counts per million) & 3.92 \\ 
  \hline
  Median dispersion & 0.03 \\ 
  \hline
  Median $\log{2}$ fold change (LFC) of genes & 0.39 \\ 
  \hline
  Median library size (sum of total counts, $\log{10}$) & 6.99 \\ 
   \hline
\end{tabular}

\caption{Description of estimated parameters}
\label{tab:Parameter-Estimation}


\end{table}

We estimated genewise dispersions parameters $\hat{\phi}_g$ and library sizes $\hat{N}_{ij}$ based on the maize RNA-seq data \citep{paschold2012complementation} through {\tt edgeR}, fit it by GLM with negative binomial distribution and log link function to get the estimated regression coefficients $\hat{\beta}_{gi}$. 

In summary, the library size (reads mapped to the transcriptome) is $\log{10}$ mean of $6.99$, the normalized median gene expressions $\log{2}$ counts per million (CPM) is $3.92$, and the median LFCs of DE genes is $0.39$. 


\subsection{Model}

In our simulated data, we used a generalized linear model (GLM) with negative binomial distribution shown in Section 2.2. For the dataset of two gene varieties, the read counts for a particular gene $g$ in variety $i$ replicate $j$ were modeled by Equation \eqref{eq:1}, where $\phi_g$ is assigned to be the estimated genewise dispersion $\hat{\phi}_g$, and the mean parameter $\mu_{gij}$ is assigned to be $\tilde{N}_{ij}\times \exp(x_i \hat{\beta}_{gi}) = \tilde{N}_{ij}\times \exp(\hat{\lambda}_{gi})$, where $\tilde{N}_{ij} \sim Unif(\min(\hat{N}_{ij}), \max(\hat{N}_{ij})); i=1,2; j = 1,2,..., nSample/2$


\subsection{Simulation Scenario}

To compare the \texttt{eBayes} method with the alternative methods across a variety of reasonable scenarios, we created several options to make up each scenario. We set up the simulation scenarios as the following Table \ref{tab:Scenario}:

\begin{table}[H]
\centering
\begin{tabular}{|r|r|r|r|r|}
\hline
sc & nGenes & nSamples & pDiff(\%) \\ 
\hline
1 & 10000 & 8 & 10 \\ 
\hline
2 & 10000 & 8 & 30 \\ 
\hline
3 & 10000 & 8 & 1 \\
\hline
4 & 10000 & 4 & 10 \\
\hline
5 & 10000 & 4 & 30 \\
\hline
6 & 10000 & 4 & 1 \\ 
\hline
7 & 10000 & 16 & 10 \\
\hline
8 & 10000 & 16 & 30 \\ 
\hline
9 & 10000 & 16 & 1 \\
\hline
10& 1000 & 8 & 10 \\
\hline
11 & 1000 & 8 & 30 \\
\hline
12 & 1000 & 8 & 1 \\ 
\hline
13 & 1000 & 4 & 10 \\
\hline
14 & 1000 & 4 & 30 \\
\hline
15 & 1000 & 4 & 1 \\ 
\hline
16 & 1000 & 16 & 10 \\
\hline
17 & 1000 & 16 & 30 \\ 
\hline
18 & 1000 & 16 & 1 \\ 
\hline
\end{tabular}
\caption{Simulation Scenario Table}
\label{tab:Scenario}
\end{table}


For a simulated count data, the number of true DE genes was determined by the proportion of differential gene expression (pDiff) and the total genes (nGenes) in the simulation scenario setup. Among the randomly selected nGenes genes, we randomly chose $nGenes \times (1-pDiff)$ genes, assigned the relative abundance of these chosen genes in two varieties the same as $\lambda_{g1} = \lambda_{g2} = 1/2\times(\hat{\beta}_{g1} + \hat{\beta}_{g2})$, where $\hat{\beta}_{gi}, i=1,2$ were the estimated regression coefficients.

We simulated nGenes $\in (10000, 1000)$ total number of genes following negative binomial distribution. The mean and dispersions were drawn from the joint distribution of means and gene-wise dispersion estimates from the real maize data \citep{paschold2012complementation} shown in Section 3.1.2. These simulated datasets were of varying total sample size nSamples $\in {(4,8,16)}$, and the samples were split into two equal-sized groups. pDiff $\in {(10\%, 30\%, 1\%)}$ of genes are true DE genes. For each scenario, we simulated $nRep = 5$ datasets with different seeds.









