% Template file for a standard thesis
\documentclass[11pt]{isuthesis}
\usepackage[pdftex]{graphicx}
%\usepackage[utf8]{inputenc}
%\usepackage{float}
%\usepackage{graphicx}
%\usepackage[english]{babel}
\usepackage{amsmath}
\usepackage{float}

% Standard, old-style thesis
\usepackage{isutraditional}   \chaptertitle
\usepackage{comment}

% Old-style, thesis numbering down to subsubsection
\alternate
\usepackage{rotating}


% Bibliography without numbers or labels
\usepackage[round]{natbib}
\bibliographystyle{plainnat}


%Optional Package to add PDF bookmarks and hypertext links
\usepackage[pdftex,hypertexnames=false,linktocpage=true]{hyperref}
\hypersetup{
	colorlinks=true,
	linkcolor=blue,
	anchorcolor=blue,
	citecolor=blue,
	filecolor=blue,
	urlcolor=blue,
	bookmarksnumbered=true,
	pdfview=FitB
}

% Use import if there are .tex files called from chapter1.tex (a table for instance) 
\usepackage{import} 

% include path for figures
\graphicspath{{figures/}}

\newenvironment{thesis}{}{}
\excludecomment{thesis} % will exclude material that is only for the thesis


\begin{document}
\DeclareGraphicsExtensions{.jpg,.pdf,.mps,.png}
\include{titlepage}
% Optional thesis dedication
\include{dedication}
% Table of Contents, List of Tables and List of Figures
\pdfbookmark[1]{TABLE OF CONTENTS}{table}
\tableofcontents
\addtocontents{toc}{\def\protect\@chapapp{}} \cleardoublepage \phantomsection
\addcontentsline{toc}{chapter}{LIST OF TABLES}
\listoftables
\cleardoublepage \phantomsection \addcontentsline{toc}{chapter}{LIST OF FIGURES}
\listoffigures
% Comment out the next line if NOT using chaptertitle
\addtocontents{toc}{\def\protect\@chapapp{CHAPTER\ }}



\cleardoublepage \phantomsection \include{acknowledgements} 
\cleardoublepage \phantomsection \include{abstract}        



\newpage
\pagenumbering{arabic}

\include{introduction}


\chapter{METHOD}

\section{Estimating the difference between read counts for a given gene}

To detemine whether the read count differences between different conditions for a given gene are greater than expected by chance, differential gene expression (DGE) tools must find a way to estimate that difference \citep{dundar2015introduction}.The two basic tasks of all DGE tools are: (1) Estimate the magnitude of differential expression between two or more conditions based on read counts from replicated samples, i.e., calculate the fold change of read counts, taking into account the differences in sequencing depth and variability; (2) Estimate the significance of the difference and correct for multiple testing. 

\section{Empirical Bayes identification of gene differential expression from RNA-seq read counts}

I consider an RNA sequencing (RNA-seq) experiment that involves two genetic varieties. For each variety, replicate RNA samples are isolated and assessed for quality. Complementary DNA (cDNA) libraries, consisting of short cDNA fragments derived from RNA, are constructed. Then, next generation sequencing technology is used to determine the {\tt reads}, in the cDNA libraries \citep{niemi2015empirical}. These reads are processed using bioinformatic algorithms to match the reads to genes. The results of read processing are summarized by a gene $\times$ sample matrix of counts. See Datta and Nettleton \citep{datta2014statistical} for more details on RNA-seq experiments and data from a statistical perspective, and see \citep{paschold2012complementation} for the biological background behind the use of RNA-seq to study gene expression differentiation. 

To use RNA-seq counts to identify genes displaing differential expression (DE), I built a hierarchical to borrow information across gene-variety means and across gene-specific overdispersion parameters, estimate the hyperparameters using an empirical Bayes procedure, and calculate empirical Bayes posterior probabilities for DE. 

\subsection{Hierarchical model for RNA-seq counts}

Let $Y_{gij}$ be the count for gene $g=1,2,..., G$, variety $i=1,2$, and replicate $j=1,2,3,...,n_i$.

I assume

\begin{equation}
\label{eq:1}
Y_{gij} \stackrel{ind}{\sim} NB(\exp(\mu_{gi}+\gamma_{ij}), \exp(\psi_g))
\end{equation}

The data model involves genewise overdispersion $\psi_g$ and a mean that depends on the gene-variety combination through $\mu_{gi}$ and on the sample through $\gamma_{ij}$. The $\mu_{gi}$ terms are of primary scientific interest. The $\gamma_{ij}$ terms are normalization factors that account for differencees in the thoroughness of sequencing from sample to sample. 

Following \citep{ji2014estimation}, we reparameterize the gene-variety mean structure into the genespecific average $\beta_{g1}$ and half-variety difference $\beta_{g2}$. For our differential expression study where number of varieties is 2, we let $i=1,2$ indicate the two varieties. The reparameterization is

\begin{equation}
\label{eq:2}
\beta_{g1} = \frac{\mu_{g1}+\mu_{g2}}{2}, \beta_{g2} = \frac{\mu_{g1}-\mu_{g2}}{2}
\end{equation}

We assume a hierarchical model for the gene-specific mean parameters and overdispersion parameters. Initially, we assume the variety averages, half-variety averages, and overdispersion parameters follow normal distributions

\begin{equation}
\label{eq:3}
\beta_{g1} \stackrel{ind}{\sim} N(\eta_{\beta_1}, \sigma^2_{\beta_1}), \beta_{g2} \stackrel{ind}{\sim} N(\eta_{\beta_2} , \sigma^2_{\beta_2}), \psi_g \stackrel{ind}{\sim} N(\eta_\psi, \sigma^2_\psi)
\end{equation}

\subsection{Empirical Bayes}

We categorize the parameters of the model into gene-specific parameters $\theta = (\theta_1, ..., \theta_G)$ where $\theta_g = (\beta_{g1}, \beta_{g2}, \psi_g)$, normalization factors $\gamma = (\gamma_{11}, ..., \gamma_{V n_V})$, and hyperparameters $\pi = (\eta, \sigma)$ where $\eta = (\eta_{\beta_1}, \eta_{\beta_2}, \eta_\psi)$ and $\sigma = (\sigma_{\beta_1}, \sigma_{\beta_2}, \sigma_\psi)$. We obtain estimates for the hyperparameters and then base gene-specific inference on the posterior conditional on these estimates \citep{niemi2015empirical}.

To obtain normalization factors $\hat{\gamma}$, we use the weighted trimmed mean of $M$ values (TMM). We use edgeR to obtain genewise dispersion estimates, $\hat{\psi}_g$, and the generalized linear model methods to obtain estimates for the remaining gene-specific parameters ($\hat{\beta}_{g1}, \hat{\beta}_{g2}$)\citep{robinson2010scaling}. Using $\hat{\theta}_g = (\hat{\beta}_{g1} , \hat{\beta}_{g2}, \hat{\psi}_g)$, we estimate hyperparameters for the location and scale parameters in the hierarchical model using a central method of moments approach. 

Conditional on the estimated normalization factors $\hat{\gamma}$ and hyperparameters $\hat{\pi}$, we perform a Bayesian analysis to re-estimate the gene-specific parameters and describe their uncertainty \citep{niemi2015empirical}. Equation \ref{eq:4} shows that conditional on $\hat{\gamma}$ and $\hat{\pi}$, the gene-specific parameters are independent and therefore conditional posterior inference across the genes can be parallelized. 

\begin{equation}
\label{eq:4}
\begin{split}
& p(\theta | y, \hat{\pi}, \hat{\gamma})  \propto \\ & \prod_{g=1}^{G} \left[ \prod_{i=1}^{2} \prod_{j=1}^{n_i} NB(y_{gij} ; \exp(\mu_{gi} + \hat{\gamma}_{ij}), \exp(\psi_g)) N(\beta_{g1} ; \hat{\eta}_{\beta_1}, \hat{\sigma}^2_{\beta_1}) p(\beta_{g2} ; \hat{\eta}_{\beta_2}, \hat{\sigma}_{\beta_2}) N(\psi_g ; \hat{\eta}_{\psi}, \hat{\sigma}^2_{\psi})  \right]
\end{split}
\end{equation}

To perform the conditional posterior inference on the gene-specific parameters, we use the statistical software Stan \citep{stan2014stan} executed through the RStan interface \citep{team2016rstan}. Stan implements a Hamiltonion Monte Carlo \citep{neal2011mcmc} to obtain samples from the posterior in equation \ref{eq:4}. 


We ran 4 simultaneous chains with random initial starting values for 1000 burn-in iterations followed by another 1000 iterations retaining every fourth sample for inference. We monitored convergence using the potential scale reduction factor and effective sample size (ESS) for $\phi_g, \alpha_g$ and $\psi_g$ \citep{gelman1992inference}. If the minimum ESS was less than 1000, we reran the chains wit double the iterations for both burn-in and inference. We continued this restarting and doubling until we obtained minimum ESS greater than 1000 for all parameters. 

\subsection{Gene expression differentiation}

In the maize context that motivates this work, we are interested in differential expression (DE), i.e. either low differential expression (LDE) or high differential expression (HDE), in gene expression. For a specific gene $g$, LDE occurs when expected expression in the second variety is less than the expected expression of first variety, i.e., $\mu_{g1} > \mu_{g2}$, or equivalently, $\beta_{g2}>0$, and HDE occurs when the expected expression in the second variety is greater than the expected expression of first variety, i.e., $\mu_{g1} < \mu_{g2}$, or equivalently, $\beta_{g2}<0$. We evaluate these probabilities based on empirical Bayes estimates of their posterior probabilities, e.g., 

\begin{equation}
\label{eq:5}
P(LDE_g | y, \hat{\pi}, \hat{\gamma}) = P(\beta_{g2}>0 | y, \hat{\pi}, \hat{\gamma}) \approx \frac{1}{M} \sum_{m=1}^M I(\beta_{g2} ^ {(m)}>0)
\end{equation}

where $\beta_{g2}^{(m)}$ is the $m^{th}$ MCMC sample from the empirical Bayes posterior.

HDE probability is defined as equation \ref{eq:6}

\begin{equation}
\label{eq:6}
P(HDE_g | y, \hat{\pi}, \hat{\gamma}) = P(\beta_{g2}<0 | y, \hat{\pi}, \hat{\gamma}) \approx \frac{1}{M} \sum_{m=1}^M I(\beta_{g2} ^ {(m)}<0)
\end{equation}

I construct a ranked list of genes according to the maximum of gene's LDE and HDE probabilities. Geneticists can use this list to prioritize future experiments to understand the molecular genetic mechanisms for differential expression \citep{niemi2015empirical}. 

I will use the term ebayes to refer to the approach defined in Sections 2.1 - 2.2 and we are assuming normal distribution for half-variety differences.

\section{Alternative DGE tools}

Among the alternative methods, {\tt sSeq}\citep{10.1093/bioinformatics/btt143}, {\tt DESeq}\citep{anders2010differential}, {\tt edgeR}\citep{robinson2007moderated}\citep{robinson2010edger}, {\tt EBSeq}\citep{leng2013ebseq}, and {\tt DESeq2}\citep{love2014moderated} assume the Negative Binomial distribution. I biefly overview these approaches. 



\section{Simulation study based on a maize experiment}

To assess the efficacy of ebayes method to identify genes demonstrating DE, I used a maize dataset with two varieties B73 and Mo17\citep{paschold2012complementation} to determine realistic parameter values for a simulation study.I compared ebayes method to approaches using the R packages {\tt sSeq, DESeq, edgeR, EBSeq, and DESeq2}.



\section{Methods Comparison Result}

Figure \ref{auc} is my facetted auc plot.

\begin{figure}[h!tb] 
\includegraphics[scale=0.4]{auc_facet_plot}
\caption{Facet AUC Plot}
\label{auc}
\end{figure}



%\appendixtitle 
%\appendix
%\include{appendix1}
%\include{appendix2}


\renewcommand{\bibname}{\centerline{BIBLIOGRAPHY}}
\unappendixtitle
\newpage
\phantomsection
%\addcontentsline{toc}{chapter}{BIBLIOGRAPHY}
\bibliographystyle{plain}
\bibliography{mybib}

\end{document}
