% Template file for a standard thesis
\documentclass[11pt]{isuthesis}
\usepackage[pdftex]{graphicx}
%\usepackage[utf8]{inputenc}
%\usepackage{float}
%\usepackage{graphicx}
%\usepackage[english]{babel}
\usepackage{amsmath}
\usepackage{float}

% Standard, old-style thesis
\usepackage{isutraditional}   \chaptertitle
\usepackage{comment}

% Old-style, thesis numbering down to subsubsection
\alternate
\usepackage{rotating}


% Bibliography without numbers or labels
\usepackage{natbib}
%\bibliographystyle{apa}


%Optional Package to add PDF bookmarks and hypertext links
\usepackage[pdftex,hypertexnames=false,linktocpage=true]{hyperref}
\hypersetup{
	colorlinks=true,
	linkcolor=blue,
	anchorcolor=blue,
	citecolor=blue,
	filecolor=blue,
	urlcolor=blue,
	bookmarksnumbered=true,
	pdfview=FitB
}

% Use import if there are .tex files called from chapter1.tex (a table for instance) 
\usepackage{import} 

% include path for figures
\graphicspath{{figures/}}

\newenvironment{thesis}{}{}
\excludecomment{thesis} % will exclude material that is only for the thesis


\begin{document}
\DeclareGraphicsExtensions{.jpg,.pdf,.mps,.png}
\include{titlepage}
% Optional thesis dedication
\include{dedication}
% Table of Contents, List of Tables and List of Figures
\pdfbookmark[1]{TABLE OF CONTENTS}{table}
\tableofcontents
\addtocontents{toc}{\def\protect\@chapapp{}} \cleardoublepage \phantomsection
\addcontentsline{toc}{chapter}{LIST OF TABLES}
\listoftables
\cleardoublepage \phantomsection \addcontentsline{toc}{chapter}{LIST OF FIGURES}
\listoffigures
% Comment out the next line if NOT using chaptertitle
\addtocontents{toc}{\def\protect\@chapapp{CHAPTER\ }}



\cleardoublepage \phantomsection \include{acknowledgements} 
\cleardoublepage \phantomsection \include{abstract}        



\newpage
\pagenumbering{arabic}

\include{introduction}


\chapter{METHOD}

\section{Estimating the difference between read counts for a given gene}

To detemine whether the read count differences between different conditions for a given gene are greater than expected by chance, differential gene expression (DGE) tools must find a way to estimate that difference \cite{dundar2015introduction}.The two basic tasks of all DGE tools are: (1) Estimate the magnitude of differential expression between two or more conditions based on read counts from replicated samples, i.e., calculate the fold change of read counts, taking into account the differences in sequencing depth and variability; (2) Estimate the significance of the difference and correct for multiple testing. 

\section{Empirical Bayes identification of gene differential expression from RNA-seq read counts}

I consider an RNA sequencing (RNA-seq) experiment that involves two genetic varieties. For each variety, replicate RNA samples are isolated and assessed for quality. Complementary DNA (cDNA) libraries, consisting of short cDNA fragments derived from RNA, are constructed. Then, next generation sequencing technology is used to determine the {\tt reads}, in the cDNA libraries \cite{niemi2015empirical}. These reads are processed using bioinformatic algorithms to match the reads to genes. The results of read processing are summarized by a gene $\times$ sample matrix of counts. 

\section{Alternative DGE tools}






\section{Methods Comparison} 
\input{figures/tab1} 

\section{Lets share an historic moment}

Figure \ref{history} is so amazing  that the rest is indeed obvious.

\begin{figure}[h!tb] \centering
\includegraphics{history}
\caption{Suarez save the day !}
\label{history}
\end{figure}



%\appendixtitle 
%\appendix
%\include{appendix1}
%\include{appendix2}


\renewcommand{\bibname}{\centerline{BIBLIOGRAPHY}}
\unappendixtitle
\newpage
\phantomsection
%\addcontentsline{toc}{chapter}{BIBLIOGRAPHY}
\bibliographystyle{plain}
\bibliography{mybib}

\end{document}
